jsf-api/src/main/java/javax/faces/component/search/SearchExpressionHandler.java

```diff
@@ -0,0 +1,289 @@
+ * and Distribution License("CDDL") (collectively, the "License").  You
+ * may not use this file except in compliance with the License.  You can
+ * obtain a copy of the License at
+ * https://glassfish.java.net/public/CDDL+GPL_1_1.html
+ * or packager/legal/LICENSE.txt.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing the software, include this License Header Notice in each
+ * file and include the License file at packager/legal/LICENSE.txt.
+ *
+ * GPL Classpath Exception:
+ * Oracle designates this particular file as subject to the "Classpath"
+ * exception as provided by Oracle in the GPL Version 2 section of the License
+ * file that accompanied this code.
+ *
+ * Modifications:
+ * If applicable, add the following below the License Header, with the fields
+ * enclosed by brackets [] replaced by your own identifying information:
+ * "Portions Copyright [year] [name of copyright owner]"
+ *
+ * Contributor(s):
+ * If you wish your version of this file to be governed by only the CDDL or
+ * only the GPL Version 2, indicate your decision by adding "[Contributor]
+ * elects to include this software in this distribution under the [CDDL or GPL
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package javax.faces.component.search;
+
+import java.util.List;
+import javax.faces.component.ContextCallback;
+import javax.faces.component.UIComponent;
+import javax.faces.context.FacesContext;
+
+/**
+ * <div class="changed_added_2_3">The <strong>SearchExpressionHandler</strong> is responsible for
+ * resolving <em>search expression(s)</em>.
+ * 
+ * <p>A <em>search expression</em> consists of either an identifier
+ * (which is matched exactly against the <code>id</code> property of
+ * a {@link UIComponent}, or an keyword (like <code>@this</code> or <code>@form</code>),
+ * or a series of such identifiers and keywords linked by
+ * the {@link javax.faces.component.UINamingContainer#getSeparatorChar} character value.
+ * The search algorithm should operates as follows, though alternate
+ * alogrithms may be used as long as the end result is the same:</p>
+ * 
+ * <ul>
+ *   <li>
+ *     Identify the {@link UIComponent} that will be the base for searching:
+ *     <ul>
+ *       <li>
+ *           If the search expression begins with the separator character
+ *           (called an "absolute" search expression),
+ *           the base will be the root {@link UIComponent} of the component

```

Comment:
This implies that it could be another component than UIViewRoot. Is this possible here?